### DISCUSSION TITLE
Installation via pip install open-webui and settings

### DISCUSSION THREAD

**Discussion by vibora** [open-webui/open-webui#2679] [Q&A]
Hi there,

I'm playing a bit with open-webui and the installation methods. I stumbled over the pip installation (beta). And I must say it works for me like a charm. Nevertheless I'm asking myself where is all the data saved when I use this installation approach? The update scripts, the chats, the settings, vector db and so on? Is this documented somewhere or can anybody help me with this question?
I'm working on a Mac with homebrew installed python 3.11.9

Thanks a lot in advance for any useful hint.

**Comment by bunnyfu** [comment]
I have successfully installed open-webui through pip on a mac. Now I'm wondering where it stores the database and downloaded models...

**Reply by vibora** [reply]
Models is pretty easy. If you use ollama then the models are stored in an .ollama folder under your user directory . Regardng the other things I still have no clue where all of this stuff is stored

**Reply by d3lay** [reply]
on my install it is under ".../.venv/Lib/site-packages/data"

**Comment by martindellavecchia** [comment]
any clue on it?. I'd like to hardcode  a negative prompt to my A1111 so I limit the kind of images generated for stable diffusion

Thanks in advance!

**Comment by shadowfaxproject** [comment]
I have similar setup: mac OS, python3.11, conda-env using anaconda3.
Used pip to install open_webui. Found following locations where data, config, icons etc are stored:
```
.../anaconda3/envs/my-conda-env/lib/python3.11/site-packages/open_webui/frontend/
.../anaconda3/envs/my-conda-env/lib/python3.11/site-packages/data/
```
If you installed it directly on the system python (i.e. not using conda or any other virtual env) then locate `.../lib/python3.11/site-packages/` first.

**Comment by en0** [comment]
I don't know if this is still useful to anyone, but I found where my data file directory was by importing the module in a python repl and just printing it to see the file path.

```
python
import open_webui
print(open_webui)
# -> /opt/homebrew/lib/python3.11/site-packages/open_webui/
exit()
ls  /opt/homebrew/lib/python3.11/site-packages/open_webui/data
# -> cache functions readme.txt tools uploads vector_db webui.db
```

**Reply by bosco025** [reply]
Thank you very much!
